### PR TITLE
VideoPress: implement "Privacy and rating" panel to v6

### LINKED ISSUE #27101: VideoPress: migrate setting privacy to v6
pxWta-1zy-p2

> I propose renaming some of the existing sections of the sidebar (particularly keeping an eye on the capitalization) so it’s more consistent with other standard blocks. 

- [ ] Privacy and rating instead of Video File Settings (so it’s easier to understand).

### PR DESCRIPTION
**Pull request by retrofox**
This PR creates and adds the new `Privacy and rating` panel, with the `Privacy` control.
The implementation registers a new `privacySetting` attribute used to render the UI smoothly but also keeps the attribute values in sync with the VideoPress video data by using the `useSyncMedia()` hook.

Fixes https://github.com/Automattic/jetpack/issues/27101

#### Changes proposed in this Pull Request:

* VideoPress: implement Privacy and rating panel

#### Testing instructions:

* Go to the block editor
* Create/edit a VideoPress video block
* Open the block sidebar
* Confirm you see the `Privacy and rating` panel
* Change the privacy value
* hard refresh
* confirm the data is persistent
* Additionally, you may like to take a look at the network tab to see how the data is synced when the post saves

https://user-images.githubusercontent.com/77539/201758884-1d65c21c-b360-4deb-83a6-cb3acbd3c6c1.mov

### CODE REVIEW

**Comment by oskosk** [comment]
* I launched a new site, and connected Jetpack VideoPress (Both Jetpack and Jetpack VideoPress were present)...
* Went to create a post, uploaded a video, waited for it to finish, and then switched to private. After saving the post (Draft) I immediately got this 
image

**Comment by retrofox** [comment]
> I launched a new site, and connected Jetpack VideoPress (Both Jetpack and Jetpack VideoPress were present)...
> Went to create a post, uploaded a video, waited for it to finish, and then switched to private. After saving the post (Draft) I immediately got this

I'm not able to reproduce it :-(, but I got something: _Please confirm you loaded the block editor in a new tab 🙏_ 

For some reason, the bridge file is not loaded when the block editor is opened in a new tab. 🍿  

https://user-images.githubusercontent.com/77539/201991070-ec7a4463-603f-4d9a-9220-7f2e4ac89387.mov

The video above shows two flows:

* I create a new post for the first one, loading the new page (the block editor) in a new tab. As you can see, the bridge file is not loaded, I have to d a hard refresh to make it load.
* The block editor opens in the same tab as the second one. The bridge is right there.

If the bridge is not loaded, then the editor won't have access to the private video.

Could you confirm this? Anyway, pretty sure it's an unrelated issue with the PR goal.

File: projects/packages/videopress/src/client/block-editor/blocks/video/components/privacy-and-rating-panel/index.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ * Sidebar Control component.
+ *
+ * @param {VideoControlProps} props - Component props.
+ * @returns {React.ReactElement}    Component template
+ */
+export default function PrivacyAndRatingPanel( { attributes, setAttributes }: VideoControlProps ) {
+	const { privacySetting } = attributes;
+
+	return (
+		<PanelBody title={ __( 'Privacy and rating', 'jetpack-videopress-pkg' ) } initialOpen={ false }>
```
Comment (nunyvega) [inline projects/packages/videopress/src/client/block-editor/blocks/video/components/privacy-and-rating-panel/index.tsx:28]: Checking other blocks [[#](https://wordpress.com/support/wordpress-editor/blocks/blog-posts-block/#block-settings)], [[#](https://wordpress.com/support/wordpress-editor/blocks/video-block/#block-toolbar)], maybe we should put the first letter in caps.
```suggestion
		<PanelBody title={ __( 'Privacy and Rating', 'jetpack-videopress-pkg' ) } initialOpen={ false }>
```

**Comment by nunyvega** [comment]
I didn't get @oskosk 's error either, testing it worked fine for me.

File: projects/packages/videopress/src/client/block-editor/blocks/video/components/privacy-and-rating-panel/index.tsx (hunk 1 above)
Comment (retrofox) [reply projects/packages/videopress/src/client/block-editor/blocks/video/components/privacy-and-rating-panel/index.tsx:28]: it was explicitly requested as that. pxWta-1zy-p2

**Comment by retrofox** [comment]
> * I launched a new site, and connected Jetpack VideoPress (Both Jetpack and Jetpack VideoPress were present)...
> * Went to create a post, uploaded a video, waited for it to finish, and then switched to private. After saving the post (Draft) I immediately got this

Addressed here https://github.com/Automattic/jetpack/pull/27435

**Review by dhasilva — APPROVED** [review]
LGTM

**Pull request merged**